src/coconut/react/macros/Setup.hx

```diff
@@ -84,21 +84,27 @@ class Setup {
           }).fields[0]);
 
         TAnonymous(ret);
-      case t:     
-        cls.pos.error('unsupported prop type $t'); 
+      case t:
+        cls.pos.error('unsupported prop type $t');
     }
 
+    var id = switch cls.isPrivate {
+      case true: macro $i{cls.name};
+      case false if  (cls.pack.concat([cls.name]).join('.') == cls.module): macro $p{cls.module.split('.')};
+      case _: macro $p{'${cls.module}.cls.name'.split('.')};
```

Comment:
should be `${cls.name}`?

Reply:
In fact I think `macro $p{'${cls.module}.${cls.name}'.split('.')}` should work in all cases. Because that is the fully qualified type name.

Reply:
Oo oops, tested the first two cases only